src/thor/bidirectional_astar.cc

```diff
@@ -312,14 +315,15 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
 
   // Find the sort cost (with A* heuristic) using the lat,lng at the
   // end node of the directed edge.
-  float dist = 0.0f;
-  float sortcost =
-      newcost.cost + astarheuristic_forward_.Get(t2->get_node_ll(meta.edge->endnode()), dist);
+  float dist_f = astarheuristic_forward_.GetDistance(t2->get_node_ll(meta.edge->endnode()));
+  float dist_r = astarheuristic_reverse_.GetDistance(t2->get_node_ll(meta.edge->endnode()));
+  float sortcost = newcost.cost +
+                   (astarheuristic_forward_.Get(dist_f) - astarheuristic_reverse_.Get(dist_r)) / 2.f;
```

Comment:
@kevinkreiser as you remember, in case of unidirectional search, we "change" edge weights in the following way:
```
dist'(u,v) = dist(u,v) + h(v) - h(u) >= 0, where h(v) - approximated time needed to get from v to the destination; and dist(u,v) is an edge weight
```

In case of bidirectional search, let's suppose we use `h_f` heuristic for forward search and `h_r` for reverse. So, our final edge weights will be:
```
dist_f(u,v) = dist(u,v) + h_f(v) - h_f(u),
dist_r(u,v) = dist(u,v) + h_r(u) - h_r(v),
```
as far as we want equal weights for the forward and reverse searches ( `dist_f(u,v) = dist_r(u,v)` ) we will get
```
 dist(u,v) + h_f(v) - h_f(u) = dist(u,v) + h_r(u) - h_r(v) -> h_f(v) + h_r(v) = h_f(u) + h_r(u) = const
```

So, now we just can easily notice that the following heuristics satisfy to conditions above:
```
h_f(u) = (approx(u,t) - approx(s,u)) / 2,
h_r(u) = -h_f(u),
where
approx(u,t) - approximated time (lower bound) needed to get from the vertex u to the target,
approx(s,u) - approximated time (lower bound) needed to get from the source to the vertex u
```